crates/nu_plugin_parse/src/nu/mod.rs

```diff
@@ -54,12 +55,45 @@ impl Plugin for Parse {
         match &input.as_string() {
             Ok(s) => {
                 let mut output = vec![];
-                for cap in self.regex.captures_iter(&s) {
+                for caps in self.regex.captures_iter(&s) {
+                    let group_count = caps.len() - 1;
+
+                    if self.column_names.len() != group_count {
+                        return Err(ShellError::labeled_error(
+                            format!(
+                                "The are {} column(s) specified: [{}], but there are only {} regex match(es): [{}]",
+                                self.column_names.len(),
+                                self.column_names.join(", "),
+                                group_count,
+                                caps.iter()
+                                    .skip(1)
+                                    .map(|m| {
+                                        if let Some(m) = m {
+                                            let m = m.as_str();
+                                            let mut m = m.replace(",","\\,");
+                                            if m.len() > 20 {
+                                                m.truncate(17);
+                                                format!("{}...", m)
+                                            } else {
+                                                m
+                                            }
+                                        } else {
+                                            "<none>".to_string()
+                                        }
+                                    })
+                                    .collect::<Vec<String>>()
+                                    .join(", ")
+                            ),
+                            "the number of regex matches does not equal the number of columns specified here",
+                            &self.pattern_tag,
+                        ));
+                    }
```

Comment:
I love the attention to detail you have in your error message 🙌 

One nitpick though: regex is an implementation detail, so I would drop mention of regex from the error messages. I noticed we have mention of regex in some other places, so I think we should drop those too (or drop our special pattern syntax and just use regexes, but I think the simplified syntax is far easier to work with).

Maybe an error message like this if we stick to our simplified pattern syntax:

```
There are {} column(s) specified in the pattern, but could only match the first {}: [{}]",
```

A short label is nice too, so I think we can make that concise:

```
could not match all columns in pattern
```

What do you think?

Reply:
Thanks for the feedback! Updated:
```
     Running `target/debug/nu --commands 'echo "tests.html" | parse "{filename}[.{ext}]?"'`
error: There are 2 column(s) specified in the pattern, but could only match the first 1: [tests.html]
- shell:1:47
1 | echo "tests.html" | parse "{filename}[.{ext}]?"
  |                           ^^^^^^^^^^^^^^^^^^^^^ could not match all columns in pattern